android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoWebExtensionExceptionTest.kt

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.browser.engine.gecko.webextension.GeckoWebExtensionException
+import mozilla.components.concept.engine.webextension.InstallationWebExtensionException
+import mozilla.components.support.test.mock
+import mozilla.components.test.ReflectionUtils
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.geckoview.WebExtension
+import org.mozilla.geckoview.WebExtension.InstallException.ErrorCodes.ERROR_USER_CANCELED
+
+@RunWith(AndroidJUnit4::class)
+class GeckoWebExtensionExceptionTest {
+
+    @Test
+    fun `GIVEN a gecko user canceled exception WHEN calling createWebExtensionException THEN return InstallationWebExtensionException UserCancelled`() {
```

Comment:
super nit: I find the GIVEN/WHEN/THEN template works well for user stories and use cases, but for implementation details like this exception wrapper its a bit awkward to read :). I think it would be fine to even omit the test, or simply `handles user cancellation` or `wraps generic gecko exception`.